src/validation.cpp

```diff
@@ -2271,7 +2271,7 @@ bool ActivateBestChain(CValidationState& state, std::shared_ptr<const CBlock> pb
 
             for (const PerBlockConnectTrace& trace : connectTrace.GetBlocksConnected()) {
                 assert(trace.pblock && trace.pindex);
-                GetMainSignals().BlockConnected(trace.pblock, trace.pindex, trace.conflictedTxs);
+                GetMainSignals().BlockConnected(trace.pblock, trace.pindex, trace.conflictedTxs, trace.pindex->nHeight);
```

Comment:
yep, that was me rushing this work a bit.

I rather prefer to go in the "good" direction and finish decoupling the wallet from the chain, and remove the `CBlockIndex*` param. At the end, the goal of this work is make the wallet self-sustainable, cache an object pointer that is managed by another module isn't what should be doing anyway.

 Just need to pass the hash and the block time instead of the block index pointer inside `BlockConnected` and cache them in the wallet. Will add it. At this work stage, isn't something significant.